nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitAndToAttribute.java

```diff
@@ -0,0 +1,191 @@
+        }
+
+        final RecordReaderFactory readerFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+        final RecordSetWriterFactory writerFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+
+        final List<FlowFile> splits = new ArrayList<>();
+        final Map<String, String> originalAttributes = original.getAttributes();
+        final String fragmentId = UUID.randomUUID().toString();
+        try {
+            session.read(original, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream in) throws IOException {
+                    try (final RecordReader reader = readerFactory.createRecordReader(originalAttributes, in, original.getSize(), getLogger())) {
+
+                        final RecordSchema schema = writerFactory.getSchema(originalAttributes, reader.getSchema());
+
+                        final RecordSet recordSet = reader.createRecordSet();
+                        final PushBackRecordSet pushbackSet = new PushBackRecordSet(recordSet);
+
+                        int fragmentIndex = 0;
+                        while (pushbackSet.isAnotherRecord()) {
+                            FlowFile split = session.create(original);
+
+                            try {
+                                final Map<String, String> attributes = new HashMap<>();
+                                final WriteResult writeResult;
+
+                                try (final OutputStream out = session.write(split);
+                                     final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, out, split)) {
+                                    final Record record = pushbackSet.next();
+                                    writeResult = writer.write(record);
+
+                                    attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                                    attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                                    attributes.put(FRAGMENT_INDEX, String.valueOf(fragmentIndex));
+                                    attributes.put(FRAGMENT_ID, fragmentId);
+                                    attributes.put(SEGMENT_ORIGINAL_FILENAME, original.getAttribute(CoreAttributes.FILENAME.key()));
+                                    attributes.putAll(writeResult.getAttributes());
+
+                                    for (RecordField recordField : schema.getFields()) {
+                                        final String fieldName = recordField.getFieldName();
+
+                                        Object value = record.getValue(recordField);
+                                        System.out.println(fieldName + value);
+                                        if (value != null) {
+                                            attributes.put(fieldName, value.toString());
```

Comment:
Are there concerns with the length of the value here?  Record fields with large values shouldn't be copied to attributes; that sort of value should remain as flowfile content.